siad/api.go

```diff
@@ -62,56 +64,31 @@ func (d *daemon) stopHandler(w http.ResponseWriter, req *http.Request) {
 
 func (d *daemon) syncHandler(w http.ResponseWriter, req *http.Request) {
 	// TODO: don't spawn multiple CatchUps
-	// TODO: return error if no peers exist
+	if len(d.core.AddressBook()) == 0 {
+		http.Error(w, "No peers available for syncing", 500)
+		return
+	}
+
 	go d.core.CatchUp(d.core.RandomPeer())
-	fmt.Fprint(w, "Sync initiated")
 }
 
-func (d *daemon) peerHandler(w http.ResponseWriter, req *http.Request) {
-	addr := network.Address(req.FormValue("addr"))
-	switch req.FormValue("action") {
-	case "add":
-		d.core.AddPeer(addr)
-		fmt.Fprintf(w, "Added %s", req.FormValue("addr"))
-	case "remove":
-		d.core.RemovePeer(addr)
-		fmt.Fprintf(w, "Removed %s", req.FormValue("addr"))
-	default:
-		http.Error(w, "Invalid peer action", 400)
-	}
+func (d *daemon) peerAddHandler(w http.ResponseWriter, req *http.Request) {
+	// TODO: this should return an error
+	d.core.AddPeer(network.Address(req.FormValue("addr")))
+}
+
+func (d *daemon) peerRemoveHandler(w http.ResponseWriter, req *http.Request) {
+	// TODO: this should return an error
+	d.core.RemovePeer(network.Address(req.FormValue("addr")))
 }
 
 func (d *daemon) hostHandler(w http.ResponseWriter, req *http.Request) {
 	// Create all of the variables that get scanned in.
-	// var ipAddress network.Address
 	var totalStorage int64
 	var minFilesize, maxFilesize, minTolerance uint64
 	var minDuration, maxDuration, minWindow, maxWindow, freezeDuration consensus.BlockHeight
 	var price, burn, freezeCoins consensus.Currency
-	var coinAddress consensus.CoinAddress
-
-	// Get the ip address.
-	// ipAddress = network.Address(req.FormValue("ipaddress"))
 
-	// The address can be either a coin address or a friend name
-	caString := req.FormValue("coinaddress")
-	// if ca, ok := e.friends[caString]; ok {
-	//	coinAddress = ca
-	// } else
-	if len(caString) != 64 {
-		http.Error(w, "Friend not found (or malformed coin address)", 400)
-		return
-	} else {
-		var coinAddressBytes []byte
-		_, err := fmt.Sscanf(caString, "%x", &coinAddressBytes)
-		if err != nil {
-			http.Error(w, "Malformed coin address", 400)
-			return
-		}
-		copy(coinAddress[:], coinAddressBytes)
-	}
-
-	// other vars require no special parsing
 	qsVars := map[string]interface{}{
```

Comment:
qs again is opaque, and in this case there's nothing indicating what it might stand for.